### PR TITLE
Run manifest check only aganist the changed manifest

### PR DESCRIPTION
**Pull request by hitenshKharva**
### Description
Run manifest check only aganist the changed manifest

### Issues Resolved
closes https://github.com/opensearch-project/opensearch-build/issues/4075

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### Testing Screenshots:
Getting list of changed files in `'manifests/**/*.yml'`
image

Running _manifest-checks-jdk11_  jobs:
image

### CODE REVIEW

**Comment by gaiksaya** [comment]
Hey @hitenshKharva ,

Thanks for the PR. This will save us a lot of resources and time. Were you able to test this workflow?
You can do that by raising a pull request against you branch `enhancement/Run-manifest-check-only-aganist-the-changed-manifest`. The PR can be adding a a dummy component to any 1 manifest file. Click on actions and see that only checks run only for that manifest unlike https://github.com/opensearch-project/opensearch-build/actions/runs/6424012688/job/17443911911

**Comment by hitenshKharva** [comment]
hello @gaiksaya 
I tried testing it but I guess I am still facing some issues with it. Trying to figure a way out creating a pull request on my branch as discussed.

File: .github/workflows/manifests.yml [hunk 1]
```diff
@@ -11,34 +11,34 @@ on:
     - cron: 0 0 * * *
 
 jobs:
-  list-manifests11:
+  list-changed-manifests:
```
[the remark this answers is not in the document]
Comment (peterzhuamazon) [reply .github/workflows/manifests.yml:14]: We are using the cron to check daily whether there is a new version.
I dont remember if we still need that for now.

[the remark this answers is not in the document]
Comment (gaiksaya) [reply .github/workflows/manifests.yml:14]: This workflow does not create PR. That workflow has been moved to jenkins https://github.com/opensearch-project/opensearch-build/blob/main/jenkins/manifests-update.jenkinsFile
I think once the cronjob is removed, we can move forward. @hitenshKharva

[the remark this answers is not in the document]
Comment (gaiksaya) [reply .github/workflows/manifests.yml:14]: Hi @hitenshKharva,

Can you remove the cronjob?
Thanks!

[the remark this answers is not in the document]
Comment (hitenshKharva) [reply .github/workflows/manifests.yml:12]: Removed the cron job. Thanks for the review.

**Review by gaiksaya — APPROVED** [review]

**Comment by gaiksaya** [comment]
Changed the PR title to match the exact change.

**Comment by gaiksaya** [comment]
Hi @hitenshKharva I see push as a trigger too. Can you try merging your test PR in your fork for which you posted above snippets. Not sure if files changed will be grabbed for `push` as a trigger.
If not we can remove that I believe, WDYT @prudhvigodithi @peterzhuamazon ?

**Comment by prudhvigodithi** [comment]
Yes @gaiksaya  we can remove the push trigger, here is the example one https://github.com/opensearch-project/opensearch-build/actions/runs/6511391743 that was triggered after the code is merged to `main` which is not required as we have one on raising a PR `pull_request`. Thank you

**Pull request merged**